### PR TITLE
executor: fix the fail of triggering topn fallback action

### LINKED ISSUE #56185: TopN can't trigger fallback action
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### PR DESCRIPTION
**Pull request by xzhangxian1008**
### What problem does this PR solve?

Issue Number: close #56185

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
fix the issue that fails to trigger the fallback action of topn executor
```

### CODE REVIEW

File: pkg/executor/sortexec/sort_spill.go [hunk 1]
```diff
@@ -101,7 +101,13 @@ func (s *parallelSortSpillAction) Action(t *memory.Tracker) {
 	}
 
 	if t.CheckExceed() && !hasEnoughDataToSpill(s.spillHelper.sortExec.memTracker, t) {
-		s.GetFallback().Action(t)
+		s.triggerFallBackAction(t)
+	}
+}
+
+func (s *parallelSortSpillAction) triggerFallBackAction(tracker *memory.Tracker) {
```
Comment (windtalker) [inline pkg/executor/sortexec/sort_spill.go:108]: can we define this `triggerFallBackAction` in the base action, so there is no need to define it in both `parallelSortSpillAction` and `topNSpillAction`

Comment (xzhangxian1008) [reply pkg/executor/sortexec/sort_spill.go:108]: > can we define this `triggerFallBackAction` in the base action, so there is no need to define it in both `parallelSortSpillAction` and `topNSpillAction`

done

**Review by windtalker — APPROVED** [review]
lgtm

**Review by wshwsh12 — APPROVED** [review]

**Pull request merged**